### PR TITLE
CLDR-10478 Add latest data from Macau census

### PR DESCRIPTION
**Pull request by conradarcturus**
This updates the Macau population language data with up-to-date census information.

CLDR-10478

- [x] This PR completes the ticket.

In order to build the auto-generated files, use ` mvn package -DskipTests=true &&  java -jar tools/cldr-code/target/cldr-code.jar ConvertLanguageData &&  java -jar tools/cldr-code/target/cldr-code.jar GenerateLikelySubtags`

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by conradarcturus** [comment]
The only error affects the whole repo " (TestAll.java:165)  Error: (TestAll.java:165) java.lang.IllegalArgumentException: Unknown system zone id: Etc/Unknown" -- not related to this change

File: common/supplemental/supplementalData.xml [hunk 1]
```diff
@@ -5818,5 +5819,6 @@ XXX Code for transations where no currency is involved
 		<reference type="R1334" uri="https://en.wikipedia.org/wiki/Laz_people#cite_note-ethnologue-1">[missing]</reference>
 		<reference type="R1335" uri="https://en.wikipedia.org/wiki/Greeks_in_Russia_and_Ukraine#cite_ref-15">Greek population in Russia -- most ancestrally used Pontic Greek -- modern usage almost certainly has dropped off but we don't have clear statistics on current usage.</reference>
 		<reference type="R1336" uri="https://joshuaproject.net/people_groups/14444/TU">[missing]</reference>
+		<reference type="R1337" uri="https://stat.gov.pl/spisy-powszechne/nsp-2021/nsp-2021-wyniki-ostateczne/">Languages spoken at home in the 2021 Polish Census</reference>
```
Comment (srl295) [inline common/supplemental/supplementalData.xml:5822]: the Polish census covers Macao?

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv [hunk 2]
```diff
@@ -787,10 +787,13 @@ Luxembourg	LU	"605,764"	100%	"62,110,000,000"	official	French	fr	87%
...
+Macao SAR China	MO	"682,070"	96%	"71,820,000,000"		English	en	22.7%			https://www.dsec.gov.mo/getAttachment/6cb29f2f-524a-488f-aed3-4d7207bb109e/E_CEN_PUB_2021_Y.aspx 2021 Census, counting people who are fluent in the language
+Macao SAR China	MO	"682,070"	96%	"71,820,000,000"	official	Portuguese	pt	2.3%			https://www.dsec.gov.mo/getAttachment/6cb29f2f-524a-488f-aed3-4d7207bb109e/E_CEN_PUB_2021_Y.aspx 2021 Census, counting people who are fluent in the language
+Macao SAR China	MO	"682,070"	96%	"71,820,000,000"	de_facto_official	Cantonese	yue	86.2%			https://www.dsec.gov.mo/getAttachment/6cb29f2f-524a-488f-aed3-4d7207bb109e/E_CEN_PUB_2021_Y.aspx 2021 Census, counting people who are fluent in the language
+Macao SAR China	MO	"682,070"	96%	"71,820,000,000"		Filipino	fil	"20,879"			https://www.dsec.gov.mo/getAttachment/6cb29f2f-524a-488f-aed3-4d7207bb109e/E_CEN_PUB_2021_Y.aspx 2021 Census, counting people who are fluent in the language
+Macao SAR China	MO	"682,070"	96%	"71,820,000,000"		Hokkien	nan	3.7%			https://www.dsec.gov.mo/getAttachment/7a3b17c2-22cc-4197-9bd5-ccc6eec388a2/E_CEN_PUB_2011_Y.aspx 2011 Census -- the language is not distinguished in the 2021 census
```
Comment (srl295) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv:796]: is this the reference intended?

Comment (conradarcturus) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/country_language_population.tsv:800]: Yes -- do you want me to re-write it?

Hokkien is a distinguished language in the 2011 Census but it does not appear in the 2021 Census for Macao.

**Review by btangmu — APPROVED** [review]

**Pull request merged**